### ISSUE TITLE
DirectAdmin Nginx Unit for Python Applications

### ISSUE THREAD

**Issue by hosseinfani** [nginx/unit#766]
Hi there,
I'm using nginx unit available in the "Advanced Feature" panel in DirectAdmin. 

I created zzzz folder to host my python app inside the public_html folder like 

/home/xxxx/domains/yyyy/public_html/zzzz/

and put the following sample python code and I make it readable and executable:

```
def application(environ, start_response):
    start_response("200 OK", [("Content-Type", "text/plain")])
    return (b"Hello, Python on Unit!")
```

I created an application:
![image](https://user-images.githubusercontent.com/8619934/193489351-f0ebd628-5da0-4223-8c01-2c67d15c8335.png)

![image](https://user-images.githubusercontent.com/8619934/193488776-e6d445f2-ebb1-48b2-a483-ad01af63bf15.png)

When I want to create a route:
![image](https://user-images.githubusercontent.com/8619934/193489169-2fd7390a-b726-46ec-95d5-e029b78ad528.png)

there is no template for python. Also, I have no idea what to put in the JSON part:
![image](https://user-images.githubusercontent.com/8619934/193489245-0c915e89-9ce1-400a-81a1-79ba7ecf25b1.png)

And when I click on create, it raises this error:
![image](https://user-images.githubusercontent.com/8619934/193488976-bb250ed7-0544-4f23-9b69-a095b687ec09.png)

**Comment by tippexs** [comment]
As the AdminPanel itself is not our software I can not say how they did implement Unit under the hood. I have found this link.

https://tickets.directadmin.com/new_ticket.php

This should work for you. Let me know if you need any more help.

**Comment by tippexs** [comment]
Any update on this? I would like to get in touch with them to learn more about what they are doing with Unit.

**Comment by hosseinfani** [comment]
Hi @tippexs 
We could solve the issue by the help from our administrator and the test python app runs successfully.
I will post the change we did for others info soon.

**Comment by tippexs** [comment]
Close due to inactivity. Please let us know if there is anything else we can do to help.

**Comment by maherchakroun** [comment]
> Hi @tippexs We could solve the issue by the help from our administrator and the test python app runs successfully. I will post the change we did for others info soon.

@hosseinfani Could you please elaborate how you successfully installed your python/django app. I have the same issue !

**Comment by hosseinfani** [comment]
@maherchakroun 
1- I shouldn't use the `public_html` in the path for the static content as it was a shortcut to another place in storage. I had to use the actual folder path of the webapp. Something like `/home/yourwebapp`

2- In the `configuration`, the `home` path should be the virtual environment that all the python packages are installed. Somthing like `/home/yourwebapp/venv`

3- Last, in the app.py (your driver code), I did the following changes:
```
< application = app
< @app.route("/neucg/", methods=["GET", "POST"])
---
> @app.route("/", methods=["GET", "POST"])

```
